Bonobo.Git.Server/Git/GitLfs/LfsAppDataStorageProvider.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Web;
+
+namespace Bonobo.Git.Server.Git.GitLfs
+{
+    /// <summary>
+    /// Provides a simple LfsDataStorageProvider implementation where LFS objects are stored directly in the AppData folder of the Bonobo web site.
+    /// </summary>
+    public class LfsAppDataStorageProvider : ILfsDataStorageProvider
+    {
+        public string GetFileUrl(string urlScheme, string urlAuthority, string requestApplicationPath, string operation, string repositoryName, string oid, long size)
+        {
+            string url = string.Concat(
+                urlScheme,
+                "://",
+                urlAuthority,
+                "/",
+                repositoryName,
+                ".git",
+                requestApplicationPath,
+                "lfs/oid/",
+                oid);
+            return url;
+        }
+
+        private string DetermineAppDataPath()
+        {
+            string candidate1 = HttpContext.Current.Server.MapPath("~/App_Data");
+            if (System.IO.Directory.Exists(candidate1))
+                return candidate1;
+
+            string candidate2 = HttpContext.Current.Server.MapPath("~/bin/App_Data");
+            if (System.IO.Directory.Exists(candidate2))
+                return candidate2;
+
+            throw new DirectoryNotFoundException("Unable to determine path to App_Data.");
+        }
+        
+        private string DetermineFilename(string repositoryName, string oid)
+        {
+            string appDataPath = DetermineAppDataPath();
+            string firstTwo = (oid + "xx").Substring(0, 2).ToLower();
+            string secondTwo = (oid + "xxyy").Substring(2, 2).ToLower();
+            var filename = Path.Combine(appDataPath, "Lfs", repositoryName, firstTwo, secondTwo, oid);
+            return filename;
+        }
+
+        public Stream GetWriteStream(string operation, string repositoryName, string oid)
+        {
+
+            string filename = DetermineFilename(repositoryName, oid);
+            string directoryName = System.IO.Path.GetDirectoryName(filename);
```

Comment:
Really not a fan of adding stuff to the using, are you? ;)

FWIW - You're choosing _not_ to include `System.IO` in a file that deals exclusively in storage.

Reply:
I normally would.  I'll fix it.